File: services/py/src/domain/entities/timeout.py
from typing import Any


class TimeoutsDTO:
    def __init__(self, timeouts: dict[str, Any]) -> None:
        self.connect_ms = timeouts["connect_ms"]
        self.read_ms = timeouts["read_ms"]
        self.write_ms = timeouts["write_ms"]
        self.total_ms = timeouts["total_ms"]

    def to_seconds(self) -> dict[str, int]:
        return {k: v / 1000 for k, v in self.__dict__.items()}

File: services/go/src/internal/domain/entities/timeout.go
package entities

import "time"

type Timeouts struct {
	ConnectMS int `yaml:"connect_ms"`
	ReadMS    int `yaml:"read_ms"`
	WriteMS   int `yaml:"write_ms"`
	TotalMS   int `yaml:"total_ms"`
}

func (t Timeouts) ToDuration() (connect, read, write, total time.Duration) {
	return time.Duration(t.ConnectMS) * time.Millisecond,
		time.Duration(t.ReadMS) * time.Millisecond,
		time.Duration(t.WriteMS) * time.Millisecond,
		time.Duration(t.TotalMS) * time.Millisecond
}
